src/services/userService.js

```diff
@@ -0,0 +1,51 @@
+let users = [];
+let userIdCounter = 1;
+
+const resetUsers = () => {
+  users = [];
+  userIdCounter = 1;
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((person) => person.id === +id) || null;
```

Comment:
It's a good practice to explicitly convert the id to a number before making the comparison. However, it's better to ensure that the id is always a number when it's assigned, to prevent potential type-related bugs.